### DISCUSSION TITLE
Allow fallback to TOTP even when passkeys are registered

### DISCUSSION THREAD

**Discussion by vishnukvmd** [ente-io/ente#2416] [Enhancements]
Currently if a customer has configured passkeys, that will supersede any TOTP that was previously set up.

Giving an option to fallback to TOTP on devices where passkeys aren't available would make the flow simpler for customers.

**Comment by asdrubale88** [comment]
Fully agreed. When setting it up I risked being locked out. Passkey is a useful and very easy way to login, but not everything is running perfectly fine on all devices, android versions, browsers, etc. so a fallback to TOTP is a more than required option. I have personally deleted all the passkeys just because of this "enforcement".

**Comment by Damariobros** [comment]
I think this is a very good idea.

**Comment by Eternal9100** [comment]
Currently the web ui gives not hint that the TOTP is no longer functional after a passkey is generated and I only found out that I had no 2FA after deleting a passkey by accident. I am using desktop with passkeys in a Bitwarden or Yubikey but pre Android 14 phones cannot handle passkeys from third party managers and in the case of Bitwarden they are only just rolling Passkeys out to mobile after considerable problem getting it to work with Xamarin. In that case I want t fall back to TOTP They have a good solution which is to back out of the browser special tab and offer alternative methods including recovery or TOTP. You can get in with a recovery code but I don't think that's a good use for them. Keeping TOTP in parrallel is a great idea. Currently the web ui gives no hint that the TOTP is no longer functional after a passkey is generated and I only found out that I had no 2FA after deleting a passkey and then login back in with just a password.
